### PR TITLE
Document Java toolchains.

### PR DESCRIPTION
**Pull request by comius**

### CODE REVIEW

File: site/docs/bazel-and-java.md [hunk 1]
```diff
@@ -23,6 +23,37 @@ migration guide to start building your Maven projects with Bazel:
...
+By default Bazel compiles the code using a JDK downloaded from a repository 
+and it executes and tests the code with the JVM installed on the local machine.
+Bazel searches for the JVM using `JAVA_HOME` or path.
+
+The resulting binaries are compatible with locally installed JVM in both 
+class file version and system libraries, which means the resulting binaries
+depend on what is installed on the machine.
+
+In order to have a hermetic compile you can use command line flag `--java_runtime_version=remotejdk_11`.
+With flag The code will be compiled for, executed, and tested on the JVM downloaded from a remote repository.
```
Comment (katre) [inline site/docs/bazel-and-java.md:47]: Remove "With flag"

File: site/docs/bazel-and-java.md [hunk 2]
```diff
@@ -66,71 +97,100 @@ The following modules, configuration fragments, and providers will help you
...
-[`--javabase`](command-line-reference.html#flag--javabase) flags.
+local_java_repository(
+  name = "additionaljdk",            # Can be used with --java_runtime_version=additionaljdk or --java_runtime_version=11
+  version = 11,                      # Optional, if not set it is autodetected
+  java_home = "/usr/lib/jdk-15/",    # Path to directory containing bin/java
+)
+```
 
-In short,
+Example configuration of remove JVM:
```
Comment (katre) [inline site/docs/bazel-and-java.md:140]: Change "remove" to "remote".

File: site/docs/bazel-and-java.md [hunk 3]
```diff
@@ -66,71 +97,100 @@ The following modules, configuration fragments, and providers will help you
...
-You can also write your own `java_runtime` and `java_toolchain` targets. As a
-tip, use `bazel query --output=build @bazel_tools//tools/jdk:all` to see how
-the built-in runtime and toolchain targets are defined.
+Predefined configurations:
+
+- `DEFAULT_TOOLCHAIN_CONFIGURATION`: all features, supports JDK versions >= 9
+- `VANILLA_TOOLCHAIN_CONFIGURATION`: no additional features, supports all JDKs
+- `JVM8_TOOLCHAIN_CONFIGURATION`: all features, JDK version 8
+- `PREBUILT_TOOLCHAIN_CONFIGURATION`: same as default, but only use prebuilt tools (`ijar`, `singlejar`)
+- `NONPREBUILT_TOOLCHAIN_CONFIGURATION`: same as default, but all tools are build from sources
```
Comment (katre) [inline site/docs/bazel-and-java.md:194]: Change "build" to "built".

File: site/docs/bazel-and-java.md (hunk 1 above)
Comment (comius) [reply site/docs/bazel-and-java.md:47]: Done.

File: site/docs/bazel-and-java.md (hunk 2 above)
Comment (comius) [reply site/docs/bazel-and-java.md:140]: Done

File: site/docs/bazel-and-java.md (hunk 3 above)
Comment (comius) [reply site/docs/bazel-and-java.md]: Done.

**Review by katre — APPROVED** [review]

**Comment by gregestren** [comment]
Sweet!

**Pull request closed without merging**